subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java

```diff
@@ -329,6 +352,10 @@ public void onOutput(OutputEvent event) {
                 return;
             }
             this.maxWorkerCount.set(newMaxWorkerCount);
+        } else if (event instanceof EndOutputEvent) {
+            if (updateNowExecutor != null) {
+                updateNowExecutor.shutdownNow();
```

Comment:
This was changed by @lacasseio earlier in a different context to avoid a potential performance regression. See `GroupingProgressLogEventGenerator`, line 91 before the change.